src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs

```diff
@@ -545,9 +545,19 @@ private void ExtractAsRegularFile(string destinationFileName)
             {
                 Access = FileAccess.Write,
                 Mode = FileMode.CreateNew,
-                Share = FileShare.None,
-                PreallocationSize = Length,
+                Share = FileShare.None
             };
+            if (OperatingSystem.IsWindows())
+            {
+                // On Windows, PreallocationSize gives a perf improvement.
+                fileStreamOptions.PreallocationSize = Length;
+            }
+            else
+            {
+                // Restore permissions.
+                // For security, limit to ownership permissions, and respect umask (through UnixCreateMode).
+                fileStreamOptions.UnixCreateMode = Mode & (UnixFileMode)0x1FF;
```

Comment:
Can we make 0x1ff a const like this:

[https://github.com/dotnet/runtime/blob/206dccb7945aaa3f26599fbe742de9022ca7ef91/src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs#L14-L23](https://github.com/dotnet/runtime/blob/206dccb7945aaa3f26599fbe742de9022ca7ef91/src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs#L14-L23)

Should we consider adding this as an actual public API so others can take advantage of it?

Reply:
I had suggested this as an enum value in the API proposal. It was removed because it would mess up the `ToString`.